tools/iscc-sum/iscc_similarity.xml

```diff
@@ -0,0 +1,211 @@
+            </param>
+            <when value="two_files">
+                <param name="file1" type="data" format="data" label="First file"/>
+                <param name="file2" type="data" format="data" label="Second file"/>
+            </when>
+            <when value="collection">
+                <param name="file_collection" type="data_collection" collection_type="list"
+                       format="data" label="File collection to compare"/>
+            </when>
+            <when value="two_collections">
+                <param name="collection1" type="data_collection" collection_type="list" format="data"
+                       label="First collection (reference)"
+                       help="Reference collection - will generate combined ISCC"/>
+                <param name="collection2" type="data_collection" collection_type="list" format="data"
+                       label="Second collection (to compare)"
+                       help="Collection to compare against reference - will generate combined ISCC"/>
+            </when>
+        </conditional>
+        <param name="threshold" type="integer" value="12" min="0" max="256"
+               label="Similarity threshold (Hamming distance)"
+               help="Maximum Hamming distance for similarity matching. 0-5: Nearly identical, 6-12: Likely similar (default), 13-20: Probably somewhat similar"/>
+    </inputs>
+    
+    <outputs>
+        <data name="output_file" format="txt" label="${tool.name} on ${on_string}"/>
+    </outputs>
+    
+    <tests>
+        <!-- Test 1: Pairwise file comparison -->
+        <test expect_num_outputs="1">
+            <conditional name="input_type">
+                <param name="input_selector" value="two_files"/>
+                <param name="file1" value="test1.png"/>
+                <param name="file2" value="test1.png"/>
+            </conditional>
+            <param name="threshold" value="12"/>
+            <output name="output_file">
+                <assert_contents>
+                    <has_text text="~00"/>
+                </assert_contents>
+            </output>
+        </test>
+        <!-- Test 2: Single collection comparison -->
+        <test expect_num_outputs="1">
+            <conditional name="input_type">
+                <param name="input_selector" value="collection"/>
+                <param name="file_collection">
+                    <collection type="list">
+                        <element name="file1" value="test1.png"/>
+                        <element name="file2" value="test2.tiff"/>
+                        <element name="file3" value="test2.tiff"/>
+                    </collection>
+                </param>
+            </conditional>
+            <param name="threshold" value="12"/>
+            <output name="output_file">
+                <assert_contents>
+                    <has_text text="~00"/>
+                </assert_contents>
+            </output>
```

Comment:
According to line 54, the tool should `Find similar files in collection` when operating this `input_selector` set to `collection`. So I'd expect that the `output_file` is a list of files which are similar, or a sub-collection of the original collection. But since the assertions are the same as in line 94, it rather looks like `output_file` is another ISCC hash?